### PR TITLE
✨ support token based authn for cloudevents grpc option.

### PR DESCRIPTION
**Pull request by morvencao**
## Summary

This PR tries to add token based authentication for cloudevents grpc option.

## Related issue(s)

Fixes #

### CODE REVIEW

File: test/integration/cloudevents/util/cert.go [hunk 1]
```diff
@@ -155,3 +156,43 @@ func SignClientCert(caCert *x509.Certificate, caKey *rsa.PrivateKey, d time.Dura
...
+	}))
+}
+
+func WriteTokenToTempFile(token string) (string, error) {
+	// create temp file and write token to it
+	return WriteToTempFile([]byte(token))
+}
+
+// WriteToTempFile writes data to a temporary file and returns the file path.
+func WriteToTempFile(data []byte) (string, error) {
```
Comment (qiujian16) [inline test/integration/cloudevents/util/cert.go:174]: These are quite common util func, would you check if there exist one already or we could move these func to a common util package.

File: pkg/cloudevents/generic/options/grpc/options.go [hunk 2]
```diff
@@ -61,12 +66,16 @@ func BuildGRPCOptionsFromFlags(configPath string) (*GRPCOptions, error) {
 	if config.ClientCertFile != "" && config.ClientKeyFile != "" && config.CAFile == "" {
 		return nil, fmt.Errorf("setting clientCertFile and clientKeyFile requires caFile")
 	}
+	if config.TokenFile != "" && config.CAFile == "" {
+		return nil, fmt.Errorf("setting tokenFile requires caFile")
```
Comment (skeeey) [inline pkg/cloudevents/generic/options/grpc/options.go:70]: so if we use the token auth, the grpc server tls must be enabled?

Comment (morvencao) [reply pkg/cloudevents/generic/options/grpc/options.go:70]: yes, this is a requirement for gRPC, token auth must have TLS enabled.

File: test/integration/cloudevents/util/cert.go (hunk 1 above)
Comment (morvencao) [reply test/integration/cloudevents/util/cert.go]: Yes, there are many places using the same function, mostly from testing code. I’ve created a common util package and updated these findings to use this new util package.

**Comment by qiujian16** [comment]
lease @skeeey for final tag

File: pkg/util/util.go [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package util
```
Comment (skeeey) [inline pkg/util/util.go:1]: these functions are mainly used for test, could we put these to pkg/testing dir?

Comment (morvencao) [reply pkg/util/util.go]: done.

**Pull request merged**